central/sensor/service/connection/connection_impl.go

```diff
@@ -314,6 +313,17 @@ func (c *sensorConnection) handleMessage(ctx context.Context, msg *central.MsgFr
 	return c.eventPipeline.Run(ctx, msg, c)
 }
 
+func shallDedupe(msg *central.MsgFromSensor) bool {
+	// Special handling of node inventory and node indexes for Sensor version 4.6 and earlier
+	ev := msg.GetEvent()
+	if ev.GetAction() == central.ResourceAction_UNSET_ACTION_RESOURCE {
+		if ev.GetNodeInventory() != nil || ev.GetIndexReport() != nil {
+			return false
+		}
+	}
```

Comment:
We need to be careful in the future if we change the action to UPDATE because then they will be deduped again. 

Also, out of curiosity: Are NodeInventory events never supposed to be deduped?

Reply:
Not deduping Node inventories (and node indexes) is an important part of this feature. Here is why:
- We must rescan the node periodically because the disk contents may change,
- We must resend node inventories (or indexes) periodically because the vulnerabilities DB in Scanner may get updated.

We could dedupe only if node disk is unchanged and no new data about vulnerabilities arrived to scanner. Because we have no control in Sensor and Central over the latter, we shall not dedupe those messages.

Reply:
Actually, it is the second time this week (and we have Tuesday) when I answer the question why those shall not be deduped. Let me add a comment in the code with that.

Reply:
I added [6d3d6d7](https://github.com/stackrox/stackrox/pull/13606/commits/6d3d6d775ad3900503dd5f2a84dd9a771240a735) with the comment and the hardening, in which we should be also safe in case we use UPDATE action in the future.

Reply:
Thank you for the explanation and the comment!